grimoire/districts/gate.py

```diff
@@ -113,15 +119,61 @@ def add_gates(
                     )
                     gates.append(Gate(location, dir))
 
-                    build_nbt(
-                        editor=editor,
+                    moisture_func = remap_threshold_high(
```

Comment:
large code block reused multiple times should move to another function (build_gates?)